### PR TITLE
feat(notifications)  : add contact groups field  to the notification panel

### PR DESCRIPTION
**Pull request by Med1233**
## Description

**Fixes** #[ (issue)](https://centreon.atlassian.net/browse/MON-20361)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/FormInputs/useFormInputs.tsx [hunk 1]
```diff
@@ -179,9 +183,42 @@ const useFormInputs = ({ panelWidth }: Props): object => {
...
+      additionalLabelClassName: classes.additionalLabel,
+      fieldName: '',
+      grid: {
+        columns: [
+          {
+            connectedAutocomplete: {
+              additionalConditionParameters: [],
+              endpoint: contactsGroupsEndpoint
+            },
+            dataTestId: 'Search contacts groups',
```
Comment (cg-tw) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/FormInputs/useFormInputs.tsx:219]:
```suggestion
            dataTestId: 'Search contact groups',
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/PanelEdit.cypress.spec.tsx [hunk 2]
```diff
@@ -332,14 +332,32 @@ describe('Edit Panel', () => {
     cy.matchImageSnapshot();
   });
 
-  it('Validates that when the Users field is empty, the user interface responds by displaying an error message and disabling the Save button', () => {
+  it('Displays the Contacts Groups field with edited notification contacts groups', () => {
```
Comment (cg-tw) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/PanelEdit.cypress.spec.tsx:335]:
```suggestion
  it('Displays the Contact Groups field with edited notification contact groups', () => {
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -15657,9 +15657,27 @@ msgstr "Les notifications sélectionnées ont été supprimées avec succès"
...
+msgstr "Rechercher des groupes de contacts"
+
+msgid "Select contacts and contact groups"
+msgstr "Sélectionner les contacts et les groupes de contacts"
+
+msgid "24h/24h - 7/7 days"
+msgstr "24h/24h - 7/7 jours"
+
+msgid "Include services for these hosts"
+msgstr "Inclure des services pour ces hôtes"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15679]:
```suggestion
msgstr "Inclure les services pour ces hôtes"
```

File: centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/api/decoders.ts [hunk 4]
```diff
@@ -75,6 +76,7 @@ export const notificationdecoder = JsonDecoder.object<NotificationType>(
   },
   'Notifications Listing',
   {
+    contactgroups: 'contactgroups',
```
Comment (Thebarda) [inline centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/api/decoders.ts:79]:
```suggestion
```

Comment (Thebarda) [reply centreon/www/front_src/src/CloudNotificationsConfiguration/EditPanel/api/decoders.ts:79]: ts.data.json will adapt keys passed in the object

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po
Comment (Thebarda) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:1]: Don't forget to add labels on others PO files

**Review by cg-tw — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**